### PR TITLE
Call graph display

### PR DESCRIPTION
**Pull request by aminelouati**
In this PR, we refactor the code and improve the visualization of the call graph. Call graphs whose diameter is greater than 4 will be displayed vertically.

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java [hunk 1]
```diff
@@ -186,31 +129,47 @@
...
+            if (!dependsOnCatalogObjects.isEmpty()) {
+                GraphNode callingCatalogObject = callGraphHolder.addNode(catalogObjectMetadata.getBucketName(),
+                                                                         catalogObjectMetadata.getName(),
+                                                                         catalogObjectMetadata.getKind(),
+                                                                         true);
+
+                for (String dependsOnCatalogObject : dependsOnCatalogObjects) {
+                    String bucketName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(0);
+                    String objectName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
+                    boolean isCatalogObjectExist = catalogObjectService.isDependsOnObjectExistInCatalog(bucketName,
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: move the variable declaration out of the "for" block

File: src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    private static final String BLACK = "#000000";
+
+    private static final int BIG_FONT = 12;
+
+    private static final int MAX_DIAMETER = 3;
+
+    @Autowired
+    private CellFactory cellFactory;
+
+    public void buildGraphPathTable(CallGraphHolder globalCallGraph, BaseTable table) {
```
Comment (mbenguig) [inline src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java]: This function could include more comments, to explain the graph building steps.

File: src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+                                                                                                           globalCallGraph.nodeSet(),
+                                                                                                           globalCallGraph.dependencySet()));
+        inspector.connectedSets().forEach(connectedComponent -> {
+            BreadthFirstIterator breadthFirstIterator = new BreadthFirstIterator(new AsSubgraph(callGraph,
+                                                                                                connectedComponent,
+                                                                                                globalCallGraph.dependencySet()));
+            Set<GraphNode> rootNodes = new HashSet<>();
+            Set<GraphNode> leafNodes = new HashSet<>();
+            while (breadthFirstIterator.hasNext()) {
+                GraphNode graphNode = ((GraphNode) breadthFirstIterator.next());
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java]: Same as my previous comment.

File: src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+            entry.getValue().stream().forEach(graphPath -> callGraphNodes.addAll(graphPath.getVertexList()));
+            Set<DefaultEdge> callGraphEdges = new HashSet<>();
+            entry.getValue().stream().forEach(graphPath -> callGraphEdges.addAll(graphPath.getEdgeList()));
+            callGraphsHashMap.put(rootNode,
+                                  new AsSubgraph(globalCallGraph.getDependencyGraph(), callGraphNodes, callGraphEdges));
+        });
+        return callGraphsHashMap;
+    }
+
+    private List<GraphPath<GraphNode, DefaultEdge>> computeAllGraphPaths(CallGraphHolder globalCallGraph) {
```
Comment (mbenguig) [inline src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java]: A little comment please !

File: src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+                                             new Image(generateBufferedImage(orderedCallGraphsPerBucket.get(mapEntry.getKey()),
+                                                                             callGraphDiameterHashMap.get((mapEntry.getKey())))).scale(350,
+                                                                                                                                       250))
+                            .scaleToFit();
+            }
+        }
+
+    }
+
+    private Map<GraphNode, Integer> computeCallGraphsDiameter(
```
Comment (mbenguig) [inline src/main/java/org/ow2/proactive/catalog/callgraph/TableGraphPathBuilder.java]: A little comment please !

File: src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java [hunk 6]
```diff
@@ -186,31 +129,47 @@
...
+                                                                                                        objectName,
+                                                                                                        WorkflowParser.LATEST_VERSION);
+                    String objectKind = isCatalogObjectExist ? catalogObjectService.getCatalogObjectMetadata(bucketName,
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java]: same same...

**Pull request merged**